### DISCUSSION TITLE
Add border_title and border_subtitle as kwargs to Widget constructor

### DISCUSSION THREAD

**Discussion by davetapley** [Textualize/textual#3807] [Ideas]
Currently you can set `border_title` like this:

https://github.com/Textualize/textual/blob/cb612395e848e4e1c35aae1e542fbaaf14bfbe03/docs/examples/guide/widgets/hello06.py#L23-L26

Or like this:

https://github.com/Textualize/textual/blob/cb612395e848e4e1c35aae1e542fbaaf14bfbe03/docs/examples/styles/border_title_colors.py#L11-L13

---

I have a third use-case where I'd like to set it on a `Container`. Inspired by [this](https://github.com/Textualize/textual/blob/cb612395e848e4e1c35aae1e542fbaaf14bfbe03/docs/examples/styles/border_sub_title_align_all.py#L6-L12) I came up with:

```py
def title_container(border_title: str, *args: Any, **kwargs: Any) -> Container:
    container = Container(*args, **kwargs)
    container.border_title = border_title
    return container
```

Use like this:
```py
        collection = title_container("Collection", classes="section")
        with collection:
            yield Select(options, allow_blank=False)
```

---

But then it occurred to me that this would (I think) be nicer:

```py
        with Collection(classes="section", border_title="Collection"):
            yield Select(options, allow_blank=False)
```

Any thoughts / concerns? Happy to have a go at a PR.

**Comment by davep** [comment]
I'm pretty sure this has come up before and we've resisted this as it adds even more keyword arguments to the already-busy constructor of every widget that defines its own `__init__`.

For what it's worth, the pattern I use is this:

```python
with Collection(classes="section") as collection:
    collection.border_title="Collection"
    yield Select(options, allow_blank=False)
```

Or, where it's been more useful, I've even taken to having a `title` utility function:

```python
with title(Collection(classes="section"), "Collection"):
    yield Select(options, allow_blank=False)
```

where `title` is something like:

```python
def title(widget: Widget, title: str) -> Widget:
    widget.border_title = title
    return widget
```

**Reply by davetapley** [reply]
Thanks! I can't believe I didn't think to do the `as` version. I like that ✅
